workshops/github_auth.py

```diff
@@ -0,0 +1,28 @@
+from django.contrib import messages
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from social.exceptions import SocialAuthBaseException
+
+from workshops.models import Person
+
+
+class NoPersonAssociatedWithGithubAccount(SocialAuthBaseException):
+    pass
+
+
+def find_user_or_abort(request, details=None, **kwargs):
+    username = details['username']
+    try:
+        user = Person.objects.get(github=username, is_active=True)
+        return {'user': user}
+    except Person.DoesNotExist:
+        # This exception is caught by GithubAuthMiddleware
+        raise NoPersonAssociatedWithGithubAccount
+
+
+class GithubAuthMiddleware(object):
```

Comment:
It looks like it's enough to inherit from `MiddlewareMixin`, but this class is available starting from Django 1.10, so first we need to upgrade Django.